solutions/silver/cf-1000C.mdx

```diff
@@ -63,35 +62,184 @@ int main() {
 		coords.push_back(c);
 		cur++;
 	}
-	vi freq(2 * n);  // This stores the frequency of a given endpoint.
+	// This stores the frequency of a given endpoint.
+	vector<int> freq(2 * n);
 	for (int i = 0; i < n; i++) {
 		/*
 		 * A segment starts from [l*, r*], so we should end the segment
 		 * at r* + 1 to ensure that r* is included into the segment.
 		 */
-		freq[compressed[segments[i].f]]++;
-		freq[compressed[segments[i].s + 1]]--;
+		freq[compressed[segments[i].first]]++;
+		freq[compressed[segments[i].second + 1]]--;
 	}
-	// This counts how many points are at a given compressed endpoint using
-	// prefix sums.
+	// prefix sums for number of points at a given compressed endpoint
```

Comment:
wow i really couldnt tell this was prefix sums!

Reply:
bruh